### DISCUSSION TITLE
Dropdown option to not close on click

### DISCUSSION THREAD

**Discussion by LasseRafn** [livewire/flux#130] [Feature Requests] [answered]
Flux supports a "Checkbox" style dropdown (https://fluxui.dev/components/dropdown#checkbox-items), but when clicking an item, the dropdown closes. 

This makes sense for actions and radios, but for checkboxes I think it would make sense to (optionally?) keep the dropdown open after checking an option, see the below video for current "issue".

Could be that there are some issues doing this? I am not sure if its very deliberate that it closes, in which case just close this 😇 

https://github.com/user-attachments/assets/a5d986b6-d77b-4b90-a764-f491a1363db3

**Comment by joshhanley** [comment]
@LasseRafn thanks for reporting! I have converted this issue to a feature request for now, so we can focus on fixing any bugs first.

**Comment by andyunleashed** [comment]
Could I also add to this feature request the option to have radio groups display as a traditional circular icon rather than a check list? The use case for this is in a dropdown on a table header where you need to filter by a particular value.

Also an awesome addition might be mutli-select.

E.g:
![CleanShot 2024-09-28 at 00 12 52](https://github.com/user-attachments/assets/64060ed5-d684-4d77-af32-f79c04ee39ec)

**Reply by joshhanley** [reply]
@andyunleashed these are separate feature requests, so please see if there is one for each of them already and if not, open a discussion for them 🙂

**Comment by dimitri-koenig** [comment]
For those who need this feature earlier then officially supported, then following is my solution:

To keep menus open after clicking, I add a "keep-open" attribute to the menu, like this:
```html
<flux:menu keep-open>
...
</flux:menu>
```

For this to work I modified the pro flux.js file, somewhere around line 4187, like this:
```js
      if (this.hasAttribute("popover") && !this.hasAttribute("keep-open")) {
        this.addEventListener("lofi-close-popovers", () => {
          setTimeout(() => this.hidePopover(), 50);
        });
      }
```

My composer patch looks like this:
```patch
diff --git a/vendor/livewire/flux-pro/dist/flux.js b/vendor/livewire/flux-pro/dist/flux.js
index 6a14a25..784d6eb 100644
--- a/vendor/livewire/flux-pro/dist/flux.js
+++ b/vendor/livewire/flux-pro/dist/flux.js
@@ -4173,7 +4173,7 @@
         }
       });
       search(this, (query) => this._focusable.focusBySearch(query));
-      if (this.hasAttribute("popover")) {
+      if (this.hasAttribute("popover") && !this.hasAttribute("keep-open")) {
         this.addEventListener("lofi-close-popovers", () => {
           setTimeout(() => this.hidePopover(), 50);
         });
```

I hope it's OK to post short snippets and patch diffs of the pro version here. Otherwise please let me know :-)

@joshhanley Would you merge this into the official package if the solution is that easy? :-)

**Reply by joshhanley** [reply]
@dimitri-koenig if that works and you can come up with a nice single word name (as @calebporzio loves them) over `keep-open` then I will PR it to Flux Pro for you 🙂

**Reply by dimitri-koenig** [reply]
@joshhanley What's wrong with "keep-open"? :-)

Let's go with "sticky", alright?

**Reply by joshhanley** [reply]
@dimitri-koenig I don't make the rules 😉

I like that

```blade
<flux:menu sticky>

</flux:menu>
```

I'm heading to bed now, but will PR sticky tomorrow if no one else comes up with any different words 🙂

**Reply by seabasss** [reply]
@joshhanley Did this get merged? Looking for the same feature :) Thanks!

**Reply by seabasss** [reply]
> I'm heading to bed now, but will PR sticky tomorrow if no one else comes up with any different words 🙂

@joshhanley Sorry to ping you – just wondering if the sticky dropdown is still in the plan or if there were issues with the solution above? Thanks!

**Reply by joshhanley** [reply]
@seabasss I never got around to it/ never ran it past Caleb so don't know if he is willing to support it, and am off for a few weeks so won't be looking at it anytime soon.

**Reply by raz-iacob** [reply]
so was a PR made with this option or I need to do it myself? I would really like to have this option

**Reply by joshhanley** [reply]
@raz-iacob no it hasn't happened yet.

**Reply by raz-iacob** [reply]
@joshhanley Do you know if it will happen?

**Reply by joshhanley** [reply]
@raz-iacob I don't know, sorry. I know Caleb talked about it in his date picker podcast, so I need to wait until I meet with him again to find out if he has implemented it or not.

**Reply by rubenlopezgea** [reply]
Any news?
I'm also looking for a popover menu with keyboard navigation and avoiding to close it on each click

**Reply by seabasss** [reply]
> Any news?
> 
> I'm also looking for a popover menu with keyboard navigation and avoiding to close it on each click

I switched out flux:menu for flux:navmenu as a workaround.

**Reply by rubenlopezgea** [reply]
> I switched out flux:menu for flux:navmenu as a workaround.

I don't want to loose keyboard control. So now we need to choose: keyboard navigation or keeping it open.
I'd love to be able to have both.

**Reply by seabasss** [reply]
Gotcha. Agreed. Not sure why this one is taking so long. Should be a one-line addition :)

**Comment by joshhanley** [comment, marked as answer]
A fix for this has been merged and will be available in the next release.

We decided to go with `keep-open` as suggested by @dimitri-koenig.

You'll be able to add `keep-open` to the menu itself or specific items, like `menu.checkbox` or `menu.item` if you want some items to close the dropdown and others to not. There will be an update in the Dropdown docs too.

**Comment by raz-iacob** [comment]
Thanks guys! That's awesome!
